eth/handler_eth.go

```diff
@@ -42,11 +43,17 @@ func (n NilPool) Get(common.Hash) *types.Transaction              { return nil }
 func (n NilPool) GetRLP(common.Hash) []byte                       { return nil }
 func (n NilPool) GetMetadata(hash common.Hash) *txpool.TxMetadata { return nil }
 
-func (h *ethHandler) TxPool() eth.TxPool {
+func (h *ethHandler) TxPool(ip netip.Addr) eth.TxPool {
 	if h.noTxGossip {
```

Comment:
we could simplify with
```suggestion
	if h.noTxGossip || (h.txGossipNetRestrict != nil && !h.txGossipNetRestrict.ContainsAddr(ip)) {
```
and then no other diff is required, or at least merge below cases to
```
	if h.txGossipNetRestrict == nil || h.txGossipNetRestrict.ContainsAddr(ip) {
		return h.txpool
	}
```